### PR TITLE
NOISSUE - Update naming of env variable

### PR DESCRIPTION
**Pull request by buraksekili**
Signed-off-by: Burak Sekili <buraksekili@gmail.com>

Pull request title should be `MF-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
This PR updates name of the `MF_ONLY_ADMIN_CREATES_USER` environment variable to `MF_USERS_ALLOW_SELF_SIGNED` that is responsible for controlling who can create a new user. 

By default, it has a `"true"` value which means everybody can create a new user. If the user sets it to another string (other than an empty string), only the admin rather than everybody can create a new user. 

### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### CODE REVIEW

File: cmd/users/main.go [hunk 1]
```diff
@@ -112,26 +112,26 @@ const (
...
+	jaegerURL      string
+	resetURL       string
+	authTLS        bool
+	authCACerts    string
+	authURL        string
+	authTimeout    time.Duration
+	adminEmail     string
+	adminPassword  string
+	passRegex      *regexp.Regexp
+	userSelfSigned string
```
Comment (dborovcanin) [inline cmd/users/main.go]: This should be bool.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: cmd/users/main.go (hunk 1 above)
Comment (buraksekili) [reply cmd/users/main.go]: okay, i will update it now

**Review by dborovcanin — APPROVED** [review]

File: cmd/users/main.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ const (
 	defAuthURL     = "localhost:8181"
 	defAuthTimeout = "1s"
 
-	defOnlyAdminCreatesUser = "" // If empty, everybody can create a user. Otherwise, only admin can create a user.
+	defUserSelfSigned = "true" // By default, everybody can create a user. Otherwise, only admin can create a user.
```
Comment (drasko) [inline cmd/users/main.go]: Instead of `selfSigned` please use `selfRegister`, so avoid confusion with signatures.

**Review by drasko — CHANGES_REQUESTED** [review]

File: cmd/users/main.go (hunk 2 above)
Comment (buraksekili) [reply cmd/users/main.go]: Updated to `selfRegister`.

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**